feat: add `assert/is-float16array`

udaykakade25 opened the pull request:
Progresses https://github.com/stdlib-js/google-summer-of-code/issues/94

## Description

> What is the purpose of this pull request?

This pull request:

-   Add `assert/is-float16array`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves itself

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

udaykakade25 committed:
feat: add assert/is-float16array

ShabiShett07 commented on lib/node_modules/@stdlib/assert/is-float16array/docs/types/index.d.ts:
```diff
@@ -0,0 +1,42 @@
+
+// EXPORTS //
+
+export = isFloat16Array;
```
```suggestion
export = isFloat16Array;

```

ShabiShett07 commented on lib/node_modules/@stdlib/assert/is-float16array/examples/index.js:
```diff
@@ -0,0 +1,91 @@
+var isFloat16Array = require( './../lib' );
+
+
+try {
```
I am not sure, but is this try catch block required. I think you have done this because certain setup do not support `Float16` am I right?

ShabiShett07 commented:
/stldib update-copyright-years

udaykakade25 committed:
Added newline at the end of code for few files

ShabiShett07 commented on lib/node_modules/@stdlib/assert/is-float16array/README.md:
```diff
@@ -0,0 +1,148 @@
+## See Also
+
+-   <span class="package-name">[`@stdlib/assert/is-float64array`][@stdlib/assert/is-float64array]</span><span class="delimiter">: </span><span class="description">test if a value is a Float64Array.</span>
+
```
```suggestion
```

ShabiShett07 commented on lib/node_modules/@stdlib/assert/is-float16array/README.md:
```diff
@@ -0,0 +1,148 @@
+
+<!-- <related-links> -->
+
+[@stdlib/assert/is-float64array]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/assert/is-float64array
```
```suggestion
[@stdlib/assert/is-float64array]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/assert/is-float64array
[@stdlib/assert/is-float32array]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/assert/is-float32array
```

ShabiShett07 commented on lib/node_modules/@stdlib/assert/is-float16array/README.md:
```diff
@@ -0,0 +1,148 @@
+bool = isFloat16Array( new Float64Array( 10 ) );
+// returns false
+
+bool = isFloat16Array( new Array( 10 ) );
```
This line causes the lint error, I am not sure with the solution, maybe removing this be a option, maybe there be a better option, not sure about it.

ShabiShett07 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

udaykakade25 committed:
Added related links for float32array in README.md

Signed-off-by: Uday Kakade <141299403+udaykakade25@users.noreply.github.com>

udaykakade25 commented:
Hello @kgryte sir,
Here's `assert/is-float16array`.
As per your directions, I skipped test cases for `test/test.js`, `examples/index.js`, and `benchmark/bechmark.js` that require float16.
Can you take a look at `docs/types`? I think the rest is implemented correctly.

ShabiShett07 commented on lib/node_modules/@stdlib/assert/is-float16array/lib/main.js:
```diff
@@ -0,0 +1,57 @@
+// MODULES //
+
+var nativeClass = require( '@stdlib/utils/native-class' );
+
```
```suggestion


```

Planeshifter commented on lib/node_modules/@stdlib/assert/is-float16array/lib/main.js:
```diff
@@ -0,0 +1,57 @@
+* @returns {boolean} boolean indicating whether value is a Float16Array
+*
+* @example
+* var Float16Array = require( '@stdlib/array/float16/lib/polyfill.js' );
```
We want to avoid loading internals of a package. Can we do `var Float16Array = require( '@stdlib/array/float16' )` here?

Planeshifter commented on lib/node_modules/@stdlib/assert/is-float16array/test/test.js:
```diff
@@ -0,0 +1,81 @@
+	'skip': ( typeof Float16Array === 'undefined' )
+};
+
+// TESTS //
```
Want two empty lines before header:

```suggestion
var isFloat16Array = require( './../lib' );


// VARIABLES //

var opts = {
	'skip': ( typeof Float16Array === 'undefined' )
};


// TESTS //
```

Planeshifter commented on lib/node_modules/@stdlib/assert/is-float16array/benchmark/benchmark.js:
```diff
@@ -0,0 +1,132 @@
+	skip: ( typeof Float16Array !== 'function' )
+};
+
+// MAIN //
```
```suggestion
var Float16Array = globalThis[ Symbol.for( '@@stdlib/array/float16' ) ];


// VARIABLES //

var opts = {
	skip: ( typeof Float16Array !== 'function' )
};


// MAIN //
```

udaykakade25 committed:
Replaced polyfill with Float16Array = require( '@stdlib/array/float16' in main.js

Signed-off-by: Uday Kakade <141299403+udaykakade25@users.noreply.github.com>

udaykakade25 committed:
Two empty lines before headers lint issue fixed in test.js

Signed-off-by: Uday Kakade <141299403+udaykakade25@users.noreply.github.com>

udaykakade25 committed:
Fixed lint erros in main.js

Signed-off-by: Uday Kakade <141299403+udaykakade25@users.noreply.github.com>

udaykakade25 committed:
Update README.md

Signed-off-by: Uday Kakade <141299403+udaykakade25@users.noreply.github.com>

udaykakade25 committed:
Update benchmark.js

Signed-off-by: Uday Kakade <141299403+udaykakade25@users.noreply.github.com>

udaykakade25 committed:
Update main.js

Signed-off-by: Uday Kakade <141299403+udaykakade25@users.noreply.github.com>

udaykakade25 committed:
Update repl.txt

Signed-off-by: Uday Kakade <141299403+udaykakade25@users.noreply.github.com>

udaykakade25 committed:
Update README.md

Signed-off-by: Uday Kakade <141299403+udaykakade25@users.noreply.github.com>

kgryte committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM. I think this package should have been added after first adding `array/float16`. Adding this prior to `array/float16` just means that we'll need to revisit this package and refactor accordingly. In the interim, I've added various placeholder TODOs.
